### DISCUSSION TITLE
How does opening browser or window works in NiceGUI?

### DISCUSSION THREAD

**Discussion by korpe1** [zauberzeug/nicegui#3640] [Q&A]
Hi, 
how and which one of my browsers nicegui open once I run my app? And where it get native windows if I enable that feature?

**Comment by rodja** [comment]
> how and which one of my browsers nicegui open once I run my app?

We simply use the [`webbrowser` module](https://docs.python.org/3/library/webbrowser.html) to open the browser url:

https://github.com/zauberzeug/nicegui/blob/a8befcd5a0e28cfb1821a2433f3e577b42a13d64/nicegui/helpers.py#L93

> And where it get native windows if I enable that feature?

That depends on your installation. We use [pywebview](https://pywebview.flowrl.com/) to open the window.
